Update calendar.md

wade-montague opened the pull request:
Make permission more prominent and reduce user hunting by providing links directly to the details of how the permissions are set in managed workflow.

# Why

The documentation was too subtle in its delivery of 'add permissions', what, where, how? 

# How

Updated docs with links to the relevant documented permissions 

# Test Plan

Preview docs and follow links.

# Checklist


- [ x ] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [ x ] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [ x ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

wade-montague committed:
Update calendar.md

Make permission more prominent and reduce user hunting by providing links directly to the details of how the permissions are set in managed workflow.

cruzach reviewed (changes requested):
can you make the same change to the SDK 41 versioned calendar.md file?

wade-montague committed:
Merge branch 'expo:master' into patch-1

wade-montague committed:
Update calendar.md

Further amends to make it clear but account for the expo-permissions depreciation as discussed : https://github.com/expo/expo/pull/12748/files#r629575604

wade-montague committed:
Update calendar.md

Further amends to make it clear but account for the expo-permissions depreciation as discussed : https://github.com/expo/expo/pull/12748/files#r629575604

wade-montague commented on docs/pages/versions/unversioned/sdk/calendar.md:
```diff
@@ -17,7 +17,7 @@ Provides an API for interacting with the device's system calendars, events, remi
 ## Configuration
 
-In managed apps, `Calendar` requires `Permissions.CALENDAR`. Interacting with reminders on iOS requires `Permissions.REMINDERS`.
+In managed apps you must add the following permissions to your app.json to use `Calendar`. Android requires [Permissions.CALENDAR](permissions#permissionscalendar) and iOS requires [Permissions.REMINDERS](permissions#permissionsreminders).
```
Thanks for the feedback, I've amended it again in both locations now taking into account your point about expo-permissions being deprecated.

cruzach commented on docs/pages/versions/unversioned/sdk/calendar.md:
```diff
@@ -17,7 +17,9 @@ Provides an API for interacting with the device's system calendars, events, remi
 ## Configuration
 
-In managed apps, `Calendar` requires `Permissions.CALENDAR`. Interacting with reminders on iOS requires `Permissions.REMINDERS`.
+In managed apps you must add following permissions to your `app.json`.
```
```suggestion
You must add following permissions to your `app.json`.
```
we can configure these permissions for you in bare as well 😄

wade-montague replied:
Good to know, I've only been working in managed work flow so don't pay too much attention to the dark side :)

cruzach commented on docs/pages/versions/unversioned/sdk/calendar.md:
```diff
@@ -17,7 +17,9 @@ Provides an API for interacting with the device's system calendars, events, remi
-In managed apps, `Calendar` requires `Permissions.CALENDAR`. Interacting with reminders on iOS requires `Permissions.REMINDERS`.
+In managed apps you must add following permissions to your `app.json`.
+- Android requires `READ_CALENDAR` & `WRITE_CALENDAR` inside the `expo.android.permissions` array.
+- IOS requires `NSRemindersUsageDescription` key added to `expo.ios.infoPlist` with a string value describing the permission request to end users.
```
```suggestion
- iOS requires the `NSRemindersUsageDescription` key be added to `expo.ios.infoPlist` with a string value describing the reason for the permission request.
```

wade-montague committed:
Update docs/pages/versions/unversioned/sdk/calendar.md

Co-authored-by: Charlie Cruzan <35579283+cruzach@users.noreply.github.com>

wade-montague committed:
Update docs/pages/versions/unversioned/sdk/calendar.md

Co-authored-by: Charlie Cruzan <35579283+cruzach@users.noreply.github.com>

wade-montague committed:
Update calendar.md

Reflective updates from unversioned/sdk/calendar.md updates.

wade-montague commented:
Committed @cruzach 's latest suggestions and reflected those changed on the sdk-41 version of the docs also.

cruzach commented on docs/pages/versions/unversioned/sdk/calendar.md:
```diff
@@ -17,7 +17,9 @@ Provides an API for interacting with the device's system calendars, events, remi
 ## Configuration
 
-In managed apps, `Calendar` requires `Permissions.CALENDAR`. Interacting with reminders on iOS requires `Permissions.REMINDERS`.
+You must add following permissions to your `app.json`.
```
```suggestion
You must add the following permissions to your `app.json`.
```

cruzach commented on docs/pages/versions/v41.0.0/sdk/calendar.md:
```diff
@@ -17,7 +17,9 @@ Provides an API for interacting with the device's system calendars, events, remi
 ## Configuration
 
-In managed apps, `Calendar` requires `Permissions.CALENDAR`. Interacting with reminders on iOS requires `Permissions.REMINDERS`.
+You must add following permissions to your `app.json`.
```
```suggestion
You must add the following permissions to your `app.json`.
```

cruzach committed:
Apply suggestions from code review
